tests/transforms/test_transform_can_be_hoisted.py

```diff
@@ -0,0 +1,138 @@
+from xdsl.builder import Builder
+from xdsl.dialects import builtin, func, scf, test
+from xdsl.dialects.arith import Addf, Addi, Constant, DivUI, Muli
+from xdsl.dialects.builtin import (
+    FloatAttr,
+    FunctionType,
+    IndexType,
+    IntegerAttr,
+    ModuleOp,
+    f32,
+    i32,
+)
+from xdsl.dialects.scf import For, Yield
+from xdsl.ir import Block, BlockArgument, Region
+from xdsl.transforms.loop_invariant_code_motion import can_be_hoisted
+
+index = IndexType()
+
+
+def test_is_loop_dependent_no_dep():
+    lb = Constant.from_int_and_width(0, i32)
+    ub = Constant.from_int_and_width(42, i32)
+    step = Constant.from_int_and_width(3, i32)
+
+    op1 = test.TestOp(result_types=[i32])
+    op2 = test.TestOp(result_types=[i32])
+
+    for_op = For(lb, ub, step, [], Block([op1, op2], arg_types=[i32]))
+
+    assert can_be_hoisted(op1, for_op.body)
+    assert can_be_hoisted(op2, for_op.body)
```

Comment:
This is great, just that there's no need for the arith ops.
```suggestion
def test_is_loop_dependent_no_dep():
    lb = test.TestOp(result_types=[i32])
    ub = test.TestOp(result_types=[i32])
    step = test.TestOp(result_types=[i32])

    op1 = test.TestOp(result_types=[i32])
    op2 = test.TestOp(result_types=[i32])

    for_op = For(lb, ub, step, [], Block([op1, op2], arg_types=[i32]))

    assert can_be_hoisted(op1, for_op.body)
    assert can_be_hoisted(op2, for_op.body)
```